*** rxvt-2.6.3/src/command.c   Mon Jun  4 23:13:34 2001
--- rxvt-2.6.3/src/command.patch.c     Mon Jun  4 23:15:11 2001
***************
*** 2378,2384 ****
      unsigned char   buf[256];
  
      va_start(arg_ptr, fmt);
!     vsprintf(buf, fmt, arg_ptr);
      va_end(arg_ptr);
      tt_write(buf, strlen(buf));
  }
--- 2378,2384 ----
      unsigned char   buf[256];
  
      va_start(arg_ptr, fmt);
!     vsnprintf(buf, sizeof(buf), fmt, arg_ptr);
      va_end(arg_ptr);
      tt_write(buf, strlen(buf));
  }
