--- rxvt-2.7.5/src/Makefile.in~	Thu Dec  7 22:21:35 2000
+++ rxvt-2.7.5/src/Makefile.in	Thu Dec  7 22:24:02 2000
@@ -114,7 +114,7 @@
 	(cd $(srcdir); $(RMF) Makefile)
 
 install: allbin alldoc
-	@$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) librxvt.la $(libdir)
+	@$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) librxvt.la $(DESTDIR)$(libdir)
 	@$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) rxvt $(DESTDIR)$(bindir)/`$(ECHO) $$p|$(SED) 's/$(EXEEXT)$$//'|$(SED) '$(transform)'|$(SED) 's/$$/$(EXEEXT)/'`;
 
 uninstall:
