--- rxvt-2.7.6/src/main.c	Mon Apr  9 18:53:19 2001
+++ rxvt-2.7.6.modified/src/main.c	Mon May 14 09:35:54 2001
@@ -1113,7 +1113,8 @@
     XPoint          spot;
     XVaNestedList   preedit_attr;
 
-    if (r->h->Input_Context == NULL
+    if (r->h->Input_Context == NULL || rxvt_IMisRunning(r) == False)
+    /* if (r->h->Input_Context == NULL
 	|| !r->TermWin.focus
 	|| !(r->h->input_style & XIMPreeditPosition)
 	|| !(r->h->event_type == KeyPress
@@ -1121,7 +1122,7 @@
 	     || r->h->event_type == SelectionNotify
 	     || r->h->event_type == ButtonRelease
 	     || r->h->event_type == FocusIn)
-	|| !rxvt_IMisRunning(r))
+	|| !rxvt_IMisRunning(r)) */
 	return;
 
     rxvt_setPosition(r, &spot);
